lightning/src/ln/channelmanager.rs

```diff
@@ -3551,33 +3562,59 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> MessageSend
 				L::Target: Logger,
 {
 	fn get_and_clear_pending_msg_events(&self) -> Vec<MessageSendEvent> {
-		//TODO: This behavior should be documented. It's non-intuitive that we query
-		// ChannelMonitors when clearing other events.
-		self.process_pending_monitor_events();
+		let result = RefCell::new(Vec::new());
+		PersistenceNotifierGuard::optionally_notify(&self.total_consistency_lock, &self.persistence_notifier, || {
+			// TODO: This behavior should be documented. It's unintuitive that we query
+			// ChannelMonitors when clearing other events.
+			self.process_pending_monitor_events();
 
-		let mut ret = Vec::new();
-		let mut channel_state = self.channel_state.lock().unwrap();
-		mem::swap(&mut ret, &mut channel_state.pending_msg_events);
-		ret
+			let mut pending_events = Vec::new();
+			let mut channel_state = self.channel_state.lock().unwrap();
+			mem::swap(&mut pending_events, &mut channel_state.pending_msg_events);
+
+			if pending_events.is_empty() {
+				NotifyOption::SkipPersist
+			} else {
+				result.replace(pending_events);
+				NotifyOption::DoPersist
```

Comment:
Because message events aren't persisted as a part of ChannelManager (and we never assume that messages will be delivered between when we create them and give them to peers), I dont think this needs a persistence. However, we should be checking if a persist is required as a result of `process_pending_monitor_events` (both here and in `process_pending_events` below).